store/tikv/mock-tikv/cluster.go

```diff
@@ -0,0 +1,282 @@
+// the Region will have no leader before calling ChangeLeader().
+func (c *Cluster) RemovePeer(regionID uint64, peerID uint64) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.regions[regionID].removePeer(peerID)
+}
+
+// ChangeLeader sets the Region's leader Peer. Caller should guarantee the Peer
+// exists.
+func (c *Cluster) ChangeLeader(regionID, leaderID uint64) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	c.regions[regionID].changeLeader(leaderID)
+}
+
+// GiveUpLeader sets the Region's leader to 0. The Region will have no leader
+// before calling ChangeLeader().
+func (c *Cluster) GiveUpLeader(regionID uint64) {
+	c.ChangeLeader(regionID, 0)
+}
+
+// Split splits a Region at the key and creates new Region.
+func (c *Cluster) Split(regionID, newRegionID uint64, key []byte, newPeerIDs []uint64, leaderPeer uint64) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	newRegion := c.regions[regionID].split(newRegionID, key, newPeerIDs, leaderPeer)
+	c.regions[newRegionID] = newRegion
+}
+
+// Region is the Region meta data.
+type Region struct {
+	meta   *metapb.Region
+	leader uint64
+}
+
+func newRegion(regionID uint64, storeIDs, peerIDs []uint64, leaderPeer uint64) *Region {
+	meta := &metapb.Region{
+		Id: proto.Uint64(regionID),
+	}
+	for i := range storeIDs {
+		meta.Peers = append(meta.Peers, &metapb.Peer{
+			Id:      proto.Uint64(peerIDs[i]),
+			StoreId: proto.Uint64(storeIDs[i]),
+		})
+	}
+	return &Region{
+		meta:   meta,
+		leader: leaderPeer,
+	}
+}
+
+func (r *Region) addPeer(storeID, peerID uint64) {
+	r.meta.Peers = append(r.meta.Peers, &metapb.Peer{
+		Id:      proto.Uint64(peerID),
+		StoreId: proto.Uint64(storeID),
+	})
+	r.incConfVer()
+}
+
+func (r *Region) removePeer(peerID uint64) {
+	for i, peer := range r.meta.Peers {
+		if peer.GetId() == peerID {
+			r.meta.Peers = append(r.meta.Peers[:i], r.meta.Peers[i+1:]...)
+			break
+		}
+	}
+	r.incConfVer()
+}
+
+func (r *Region) changeLeader(leaderID uint64) {
+	r.leader = leaderID
+}
+
+func (r *Region) split(newRegionID uint64, key []byte, newPeerIDs []uint64, leaderPeer uint64) *Region {
+	var storeIDs []uint64
+	for _, peer := range r.meta.Peers {
+		storeIDs = append(storeIDs, peer.GetStoreId())
+	}
+	region := newRegion(newRegionID, storeIDs, newPeerIDs, leaderPeer)
+	region.updateKeyRange(key, r.meta.EndKey)
+	r.updateKeyRange(r.meta.StartKey, key)
+	return region
+}
+
+func (r *Region) updateKeyRange(start, end []byte) {
+	r.meta.StartKey = start
+	r.meta.EndKey = end
+	r.incVersion()
+}
+
+func (r *Region) incConfVer() {
+	r.meta.RegionEpoch = &metapb.RegionEpoch{
+		ConfVer: proto.Uint64(r.meta.GetRegionEpoch().GetConfVer() + 1),
+		Version: r.meta.GetRegionEpoch().Version,
+	}
+}
+
+func (r *Region) incVersion() {
+	r.meta.RegionEpoch = &metapb.RegionEpoch{
+		ConfVer: r.meta.GetRegionEpoch().ConfVer,
+		Version: proto.Uint64(r.meta.GetRegionEpoch().GetVersion() + 1),
+	}
+}
+
+// Store is the Store's meta data.
+type Store struct {
```

Comment:
@coocood We may add methods such as `UpdateAddress()` in the future for test use.